crypto/fipsmodule/CMakeLists.txt

```diff
@@ -33,6 +33,45 @@ if(ARCH STREQUAL "x86_64")
     x86_64-mont5.${ASM_EXT}
     x86_64-mont.${ASM_EXT}
   )
+
+  if (UNIX AND NOT APPLE)
+    # Copy the s2n-bignum assembly files to the binary dir.
+    file(
+      COPY
+      ${PROJECT_SOURCE_DIR}/third_party/s2n-bignum/x86_att/p384/bignum_add_p384.S
+      ${PROJECT_SOURCE_DIR}/third_party/s2n-bignum/x86_att/p384/bignum_sub_p384.S
+      ${PROJECT_SOURCE_DIR}/third_party/s2n-bignum/x86_att/p384/bignum_neg_p384.S
+      ${PROJECT_SOURCE_DIR}/third_party/s2n-bignum/x86_att/p384/bignum_tomont_p384.S
+      ${PROJECT_SOURCE_DIR}/third_party/s2n-bignum/x86_att/p384/bignum_deamont_p384.S
+      ${PROJECT_SOURCE_DIR}/third_party/s2n-bignum/x86_att/p384/bignum_montmul_p384.S
+      ${PROJECT_SOURCE_DIR}/third_party/s2n-bignum/x86_att/p384/bignum_montsqr_p384.S
+      ${PROJECT_SOURCE_DIR}/third_party/s2n-bignum/x86_att/p384/bignum_nonzero_6.S
+      ${PROJECT_SOURCE_DIR}/third_party/s2n-bignum/x86_att/p384/bignum_littleendian_6.S
+      DESTINATION ${CMAKE_CURRENT_BINARY_DIR}
+    )
+
+    # We add s2n-bignum files to a separate list because they need
+    # to go through C preprocessor in case of a static build.
+    set(
+      BCM_ASM_SOURCES_S2N_BIGNUM
+
+      bignum_add_p384.S
+      bignum_sub_p384.S
+      bignum_neg_p384.S
+      bignum_tomont_p384.S
+      bignum_deamont_p384.S
+      bignum_montmul_p384.S
+      bignum_montsqr_p384.S
+      bignum_nonzero_6.S
+      bignum_littleendian_6.S
+    )
+  endif()
+
+  if(NOT FIPS_DELOCATE)
+    # If we don't have to use the delocate script then s2n-bignum files
+    # are simply added to BCM_ASM_SOURCES.
+    list(APPEND BCM_ASM_SOURCES ${BCM_ASM_SOURCES_S2N_BIGNUM})
+  endif()
```

Comment:
it works even without that because I added S2N_ASM_SOURCES to lines 351 and 381. But I'll remove those two lines and add the same `if (NOT DELOCATE)` to aarch64 part